### PR TITLE
feat(azure): Add private dns record sets and vnet links

### PR DESCRIPTION
**Pull request by yvardhineni**
#### Summary
Adds Azure Private DNS Zone record sets and VNet Links

### CODE REVIEW

File: plugins/source/azure/resources/services/privatedns/record_sets.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/v4/faker"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/gorilla/mux"
+)
+
+func recordSets() *schema.Table {
+	return &schema.Table{
+		Name:                 "azure_privatedns_record_sets",
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/privatedns/record_sets.go:18]:
```suggestion
		Name:                 "azure_privatedns_private_zone_record_sets",
```

File: plugins/source/azure/resources/services/privatedns/record_sets.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		Resolver:             fetchRecordSets,
+		PostResourceResolver: client.LowercaseIDResolver,
+		Description:          "https://learn.microsoft.com/en-us/rest/api/dns/privatedns/record-sets/list?tabs=HTTP#recordset",
+		Multiplex:            client.SubscriptionMultiplexRegisteredNamespace("azure_privatedns_record_sets", client.Namespacemicrosoft_network),
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/privatedns/record_sets.go:22]:
```suggestion
		Multiplex:            client.SubscriptionMultiplexRegisteredNamespace("azure_privatedns_private_zone_record_sets", client.Namespacemicrosoft_network),
```

File: plugins/source/azure/resources/services/privatedns/virtual_network_links.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	"github.com/cloudquery/cloudquery/plugins/source/azure/client"
+	"github.com/cloudquery/plugin-sdk/v4/faker"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/gorilla/mux"
+)
+
+func virtualNetworkLinks() *schema.Table {
+	return &schema.Table{
+		Name:                 "azure_privatedns_virtual_network_links",
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/privatedns/virtual_network_links.go:18]:
```suggestion
		Name:                 "azure_privatedns_private_zone_virtual_network_links",
```

File: plugins/source/azure/resources/services/privatedns/virtual_network_links.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		Resolver:             fetchVirtualNetworkLinks,
+		PostResourceResolver: client.LowercaseIDResolver,
+		Description:          "https://learn.microsoft.com/en-us/rest/api/dns/privatedns/virtual-network-links/list?tabs=HTTP#virtualnetworklink",
+		Multiplex:            client.SubscriptionMultiplexRegisteredNamespace("azure_privatedns_virtual_network_links", client.Namespacemicrosoft_network),
```
Comment (erezrokah) [inline plugins/source/azure/resources/services/privatedns/virtual_network_links.go:22]:
```suggestion
		Multiplex:            client.SubscriptionMultiplexRegisteredNamespace("azure_privatedns_private_zone_virtual_network_links", client.Namespacemicrosoft_network),
```

**Review by erezrokah — COMMENTED** [review]
Thanks for the contribution @yvardhineni 🚀 Looks great.
Added a few comments regarding the table names. We usually include the parent table name (in singular) in the child's name, see example https://github.com/cloudquery/cloudquery/pull/12410/files#diff-d891ecd9395ed7888be8d30554ec319423e7b4821ca871b81235496fab2a5045R14.

This makes it easier to understand the relation from looking at the name.
Please let me know what you think

**Comment by yvardhineni** [comment]
Agreed, some convention always help

**Review by erezrokah — APPROVED** [review]
Thanks for the follow up @yvardhineni, look great 🎸

**Pull request merged**